workers/unity/Packages/com.improbable.gdk.playerlifecycle/Systems/PlayerCreation/SendCreatePlayerRequestSystem.cs

```diff
@@ -14,75 +12,74 @@ public class SendCreatePlayerRequestSystem : ComponentSystem
     {
         private readonly EntityId playerCreatorEntityId = new EntityId(1);
 
-        private struct NewEntityData
-        {
-            public readonly int Length;
-            [ReadOnly] public ComponentDataArray<WorkerEntityTag> DenotesWorkerEntity;
-            [ReadOnly] public ComponentDataArray<OnConnected> DenotesJustConnected;
-            public EntityArray Entities;
-        }
-
-        private struct SendData
-        {
-            public readonly int Length;
-            [ReadOnly] public ComponentDataArray<PlayerCreator.CommandSenders.CreatePlayer> RequestSenders;
-            [ReadOnly] public ComponentDataArray<ShouldRequestPlayerTag> DenotesShouldRequestPlayer;
-            public EntityArray Entities;
-        }
-
-        private struct ResponseData
-        {
-            public readonly int Length;
-            [ReadOnly] public ComponentDataArray<PlayerCreator.CommandResponses.CreatePlayer> Responses;
-            [ReadOnly] public ComponentDataArray<WorkerEntityTag> DenotesWorkerEntity;
-            public EntityArray Entities;
-        }
-
+        private CommandSystem commandSystem;
+        private WorkerSystem workerSystem;
         private ILogDispatcher logDispatcher;
 
-        [Inject] private NewEntityData newEntityData;
-        [Inject] private SendData sendData;
-        [Inject] private ResponseData responseData;
+        private ComponentGroup initializationGroup;
+        private ComponentGroup playerSpawnGroup;
 
         protected override void OnCreateManager()
         {
             base.OnCreateManager();
 
-            logDispatcher = World.GetExistingManager<WorkerSystem>().LogDispatcher;
+            workerSystem = World.GetExistingManager<WorkerSystem>();
+            commandSystem = World.GetExistingManager<CommandSystem>();
+            logDispatcher = workerSystem.LogDispatcher;
+
+            initializationGroup = GetComponentGroup(
+                ComponentType.ReadOnly<WorkerEntityTag>(),
+                ComponentType.ReadOnly<OnConnected>()
+            );
+
+            playerSpawnGroup = GetComponentGroup(
+                ComponentType.ReadOnly<ShouldRequestPlayerTag>()
+            );
         }
 
         protected override void OnUpdate()
         {
-            for (int i = 0; i < newEntityData.Length; ++i)
+            if (!initializationGroup.IsEmptyIgnoreFilter)
```

Comment:
Didn't we find out this was somewhat expensive? I can't exactly recall

Reply:
I'll double check, but I remember `Length` doing this under the hood anyways.

Reply:
Did a check. `IsEmptyIgnoreFilter` will go through all matchingt archetypes until it hits an archetype that is NOT empty. This means that most of the times it will go through all archetypes.
The `GetEntityArray` function that we try to avoid by doing this check, also goes through all matching archetypes to count the amount of entities it matches. This makes the early out rather useless since we only grab 1 Array. Removing the check.